hudi-utilities/src/main/java/org/apache/hudi/utilities/config/HoodieDeltaStreamerConfig.java

```diff
@@ -0,0 +1,283 @@
+
+    public static final ConfigProperty<String> TARGET_SCHEMA_TABLE_PROP = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "target.schema.hive.table")
+        .noDefaultValue()
+        .withDocumentation("Hive table from where target schema can be fetched");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_CONNECTION_URL = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.connection.url")
+        .noDefaultValue()
+        .withDocumentation("The JDBC URL to connect to. The source-specific connection properties may be specified in the URL."
+            + " e.g., jdbc:postgresql://localhost/test?user=fred&password=secret");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_DRIVER_TYPE = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.driver.type")
+        .noDefaultValue()
+        .withDocumentation("The class name of the JDBC driver to use to connect to this URL. e.g. org.h2.Driver");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_USERNAME = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.username")
+        .noDefaultValue()
+        .withDocumentation("Username for the connection e.g. fred");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_PASSWORD = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.password")
+        .noDefaultValue()
+        .withDocumentation("Password for the connection e.g. secret");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_DBTABLE = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.dbtable")
+        .noDefaultValue()
+        .withDocumentation("The table with the schema to reference e.g. test_database.test1_table or test1_table");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_TIMEOUT = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.timeout")
+        .noDefaultValue()
+        .withDocumentation("The number of seconds the driver will wait for a Statement object to execute. Zero means there is no limit. "
+            + "In the write path, this option depends on how JDBC drivers implement the API setQueryTimeout, e.g., the h2 JDBC driver "
+            + "checks the timeout of each query instead of an entire JDBC batch. It defaults to 0.");
+
+    public static final ConfigProperty<String> SOURCE_SCHEMA_JDBC_NULLABLE = ConfigProperty
+        .key(DELTA_STREAMER_SCHEMAPROVIDER_CONFIG_PREFIX + "source.schema.jdbc.nullable")
+        .noDefaultValue()
+        .withDocumentation("If true, all the columns are nullable.");
+
+    private static final ConfigProperty<String> PROTO_SCHEMA_PROVIDER_PREFIX = ConfigProperty
```

Comment:
Similar for `ProtoClassBasedSchemaProvider` configs

Reply:
I would say that for configs specific to a particular schema provider, separate them into a new class, e.g., `ProtoClassBasedSchemaProviderConfig`, so we can have separate config subsection for each schema provider in the configuration page.